siliconcompiler/dependency.py

```diff
@@ -0,0 +1,170 @@
+import os
+import requests
+import tarfile
+from pathlib import Path
+import importlib
+from git import Repo, GitCommandError
+from urllib.parse import urlparse
+import shutil
+from time import sleep
+
+
+def path(chip, package):
+    """
+    Compute dependency data path
+    Additionally cache dependency data if possible
+    Parameters:
+    arg1 (package): Package with dependency source info
+        package.dependency (dict):
+            Mandatory keys:
+            'path': '/path/on/network/drive',
+            or
+            'path': 'file:///path/on/network/drive',
+            or
+            'path': e.g. 'git+https://github.com/xyz/xyz/'
+            'commitid': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866',
+            'name': 'your-dependency'
+            or
+            'path': e.g. 'git://github.com/xyz/xyz/'
+            'commitid': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866',
+            'name': 'your-dependency'
+            or
+            'path': e.g. 'git+ssh://github.com/xyz/xyz/'
+            'commitid': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866',
+            'name': 'your-dependency'
+            or
+            'path': e.g. 'ssh://github.com/xyz/xyz/'
+            'commitid': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866',
+            'name': 'your-dependency'
+            or
+            'path': e.g. 'https://github.com/xyz/xyz/archive/',
+            'commitid': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866',
+            'name': 'your-dependency'
+            or
+            'path': e.g. 'https://zeroasic.com/xyz.tar.gz',
+            'name': e.g. 'your-dependency-version-1'
```

Comment:
is this just a string now?

Reply:
Yep if it just points to some random tar file on the internet.
In that case we don't have any chance to verify the `name` or `commitid`.
We still need to have a name of where we want to store the cached files.